File: orion-viewer/src/universe/constellation/orion/viewer/device/OnyxUtil.java
package universe.constellation.orion.viewer.device;

import java.lang.reflect.Method;

import universe.constellation.orion.viewer.Common;

/**
 * Created by mike on 9/12/14.
 */
public class OnyxUtil {

    public static boolean isEinkDevice() {
        try {
            Class epdControllerClass = Class.forName("com.onyx.android.sdk.device.DeviceInfo");

            Method singletonGetter = epdControllerClass.getDeclaredMethod("singleton");
            Object deviceInfo = singletonGetter.invoke(null);

            Method controllerGetter = epdControllerClass.getDeclaredMethod("getDeviceController");
            Object controller = controllerGetter.invoke(deviceInfo);
            Common.d("Onyx controller is " + controller);

            Method isEInkScreen = controller.getClass().getDeclaredMethod("isEInkScreen");
            Object isEinkResult = isEInkScreen.invoke(controller);
            Common.d("Onyx isEinkResult is " + isEinkResult);

            return ((Boolean) isEinkResult);
        } catch (Exception e) {
            Common.d(e);
        }
        return false;
    }

}

File: orion-viewer/src/main/java/universe/constellation/orion/viewer/device/OnyxUtil.kt
package universe.constellation.orion.viewer.device

import universe.constellation.orion.viewer.log

object OnyxUtil {

    val isEinkDevice: Boolean by lazy {
        try {
            val epdControllerClass = Class.forName("com.onyx.android.sdk.device.DeviceInfo")

            val singletonGetter = epdControllerClass.getDeclaredMethod("singleton")
            val deviceInfo = singletonGetter.invoke(null)

            val controllerGetter = epdControllerClass.getDeclaredMethod("getDeviceController")
            val controller = controllerGetter.invoke(deviceInfo)
            log("Onyx controller is $controller")

            val isEInkScreen = controller.javaClass.getDeclaredMethod("isEInkScreen")
            val isEinkResult = isEInkScreen.invoke(controller)
            log("Onyx isEinkResult is $isEinkResult")

            return@lazy isEinkResult as Boolean
        } catch (e: Exception) {
            log(e)
        }
        false
    }

}
